### PR TITLE
Using the BIT instruction in Armv8 table selection.

### PR DESCRIPTION
**Pull request by nebeid**
### Description of changes: 
Table selection function select_w5 is 35% faster than before (AND/OR combination), making P-256
ECDH, ECDSA Sign and ECDSA Verify faster by
2%, 3.7% and 1%, respectively.

| P-256, GV2   | Before (Ops/s) | After (MB/s) | Improvement |
:--------------|---------------:|-------------:|:------------:
| ECDH         |      6019      |     6140     |    1.02x    |
| ECDSA Sign   |     19028      |    19732     |    1.037x   |
| ECDSA Verify |      6595      |     6656     |    1.01x    |

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: crypto/fipsmodule/ec/asm/p256-armv8-asm.pl [hunk 1]
```diff
@@ -1583,157 +1583,111 @@
...
-    orr     $Re.16b, $Re.16b, $T0e.16b
-    orr     $Rf.16b, $Rf.16b, $T0f.16b
-
-    // Decrement loop counter; loop back if not 0
-    subs    $Ctr, $Ctr, #1                  ////    dec %rax
-    bne     .Lselect_w5_loop                ////    jnz .Lselect_loop_sse_w5
+
+    // $Mask_64 := ($Idx_ctr == $index)? All 1s : All 0s
+    cmp     $Idx_ctr, $index
+    csetm   $Mask_64, eq
```
Comment (sebpop) [inline crypto/fipsmodule/ec/asm/p256-armv8-asm.pl:1613]: `bit` instructions have no effect on the result `$Ra.16b` when `$Mask.16b` is zero.
We can stop executing the loop body as soon as we know `$Mask.16b` is zero, which is after the above `cmp $Idx_ctr, $index` instruction.
If `$Idx_ctr != $index` we could just bump the address `add $in_t, $in_t, #94` and go to next iteration.
That would remove both `ld1` instructions and all the `bit` instructions below.

File: crypto/fipsmodule/ec/asm/p256-armv8-asm.pl [hunk 2]
```diff
@@ -1583,157 +1583,111 @@
...
-    and     $T0d.16b, $T0d.16b, $TMP0.16b    ////    por    $T0c, $Rc
-    orr     $Rc.16b, $Rc.16b, $T0c.16b       ////    por    $T0d, $Rd
-    orr     $Rd.16b, $Rd.16b, $T0d.16b
-
-    // Decrement loop counter; loop back if not 0
-    subs    $Ctr, $Ctr, #1                  ////    dec %rax
-    bne     .Lselect_w7_loop                ////    jnz .Lselect_loop_sse_w7
+
+    // $Mask_64 := ($Idx_ctr == $index)? All 1s : All 0s
+    cmp     $Idx_ctr, $index
```
Comment (sebpop) [inline crypto/fipsmodule/ec/asm/p256-armv8-asm.pl:1671]: Same as above.

**Review by sebpop — COMMENTED** [review]
Maybe we could further optimize this loop?

**Review by sebpop — APPROVED** [review]

**Review by darylmartin100 — APPROVED** [review]

**Review by darylmartin100 — APPROVED** [review]

**Comment by nebeid** [comment]
The [force-push](https://github.com/awslabs/aws-lc/compare/1aa2b8ebbd67d97052a0a0a041fcfc09617eeae9..280c63947d88ab5811c9029a2f3e72b6b5cc340f) from 1aa2b8e to 280c639 includes a change to a comment in https://github.com/awslabs/aws-lc/pull/452/commits/280c63947d88ab5811c9029a2f3e72b6b5cc340f in `p256-armv8-asm.pl` in this branch and the changes in this commit (PR #450) https://github.com/awslabs/aws-lc/commit/e7af44e8390f65b5ba7f41efaab83cb63dcced07.

**Review by dkostic — APPROVED** [review]

**Pull request merged**